src/addons/addonapi.cpp

```diff
@@ -90,6 +90,22 @@ void AddonApi::initialize() {
   if (s_constructorCallback) {
     s_constructorCallback(this);
   }
+
+  m_timer.setSingleShot(true);
+}
+
+void AddonApi::setTimedCallback(int interval, const QJSValue& callback) {
+  if (!callback.isCallable()) {
+    logger.debug() << "No callback received";
+    return;
+  }
+
+  // Creates a timer for `interval` msec before executing `callback`
+  m_timer.setInterval(interval);
```

Comment:
Does the previous timer need to be stopped or does setInterval do that automatically?